packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -202,7 +210,11 @@ contract Bridge is EssentialContract, IBridge {
         uint256 refundAmount;
 
         // Process message differently based on the target address
-        if (message.to == address(this) || message.to == address(0)) {
+        if (
+            message.to == address(0) || message.to == address(this)
+                || message.to == ethVault
+        ) {
+            // || message.to == resolve("taiko", false)
```

Comment:
I removed this line: let me know if this may be a problem?

The reason to remove this one is to make Bridge sharable by multiple L2s, so there will be multiple "taiko" addresses, associated with different chainIds.